### PR TITLE
DOCSP-32887 Compass 1.39.4 Release

### PR DESCRIPTION
**Pull request by ajhuh-mdb**
## DESCRIPTION
Compass 1.39.4 Release notes.

## STAGING
https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/DOCSP-32887-1.39.4-release/release-notes/#compass-1.39.4

## JIRA
https://jira.mongodb.org/browse/DOCSP-32897

## BUILD LOG
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=64f86c0e24fcc731b4b9c4b1

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/release-notes.txt [hunk 1]
```diff
@@ -10,6 +10,26 @@ Release Notes
...
+
+*Released September 6, 2023*
+
+Bug Fixes:
+
+- Allows ``[object Object]`` as a valid string value in TypeChecker 
+  (:issue:`COMPASS-7132`).
+- Doesn't treat non-numbers in CSV headers as array indexes 
+  (:issue:`COMPASS-7157`).
+- Limits when custom paste handling is applied; uses clipboard data when
```
Comment (jeff-allen-mongo) [inline source/release-notes.txt:24]: [suggestion]

Our style guide suggests to [avoid semicolons](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#semicolons).

Maybe break this into separate sentences?

File: source/release-notes.txt [hunk 2]
```diff
@@ -10,6 +10,26 @@ Release Notes
...
+  (:issue:`COMPASS-7132`).
+- Doesn't treat non-numbers in CSV headers as array indexes 
+  (:issue:`COMPASS-7157`).
+- Limits when custom paste handling is applied; uses clipboard data when 
+  auto-fixing user input (:issue:`COMPASS-7149`).
+- Updates Electron to v24.8.2 to address security vulnerabilities. This updates 
+  specifically addresses CVE-2023-4427 and CWE-119.
+
+`Full Changelog available on GitHub
+<https://github.com/mongodb-js/compass/compare/v1.39.3...v1.39.4-beta.2>`__
```
Comment (jeff-allen-mongo) [inline source/release-notes.txt:30]: [question]

Is there a reason we're comparing with the beta branch? Can we use this URL instead?

https://github.com/mongodb-js/compass/compare/v1.39.3...v1.39.4

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM % a couple nits. Thanks @ajhuh-mdb!

File: source/release-notes.txt (hunk 2 above)
Comment (ajhuh-mdb) [reply source/release-notes.txt:30]: Ah! That was my mistake — when I created the PR, the 1.39.4 release wasn't officially out yet. Updating.

**Review by gribnoysup — APPROVED** [review]

**Pull request merged**